packages/parse5-html-rewriting-stream/package.json

```diff
@@ -18,7 +18,9 @@
     ],
     "license": "MIT",
     "main": "dist/index.js",
-    "exports": "dist/index.js",
+    "module": "dist/index.js",
+    "types": "dist/index.d.ts",
```

Comment:
by setting this, does that mean importing an exact file will fail in typescript? since it'd look for the definition in here rather than `{whatever-you-imported}.d.ts`. maybe thats your intention? still trying to understand the overall change tbh so could've just misunderstood

Reply:
With this PR, exact file imports are no longer allowed — all you can do is import the modules. This is done through the `exports` field.

As for the `types` field — from my understanding, this is for TS to figure out how to resolve a plain module import (ie. `import ... from 'parse5'`). TS should use co-located `.d.ts` files when you actually import a file.

Reply:
because we don't bundle anything, i feel like we don't need it. typescript will already detect that there's an `index.d.ts` for the `index.js`. using a bare import will resolve `main` or whatever else, and then typescript will find the parallel `d.ts` file iirc.

Reply:
I am honestly not sure if we need it or not. Eg. `tsc` might be able to pick up the type definitions, but will NPM show the TS badge if there is no `types` field in the `package.json`?

My gut instinct is to keep this field, just to be safe.

Reply:
i just tried it on a package, with `types` set using a direct import of a module. it seems to work fine, so i think you're ok to leave this here